numpy_ufuncs_plugin.py

```diff
@@ -0,0 +1,42 @@
+from mypy.nodes import ARG_POS
+from mypy.plugin import Plugin
+from mypy.types import CallableType
+
+UFUNC_ARITIES = {
```

Comment:
We would need a big dictionary of the ufuncs and their arities.

Reply:
UFuncs should provide all the information you need for this, no? Such as `np.sin.nin`, `np.sin.nout`, and `np.sin.signature` (which is None for normal ufuncs).

Reply:
I've been assuming it's bad practice to import NumPy instead of working just from the AST, but heck I don't know it seems pretty reasonable here... maybe we should just do it? Worried about triggering https://xkcd.com/292/.

Reply:
Ohh, I didn't think about that and I guess it is a mypy/typing code smell question and I have no nose for typing :). I suppose we thus need something like a decorator for ufuncs that duplicates some of the information? (Its not like we can get the decorator information to the C-level cleanly probably).
Also, it is not like NumPy is the only one with ufuncs. SciPy will need to use this hook as well.

Reply:
Switched to pulling it off the ufunc object in https://github.com/numpy/numpy-stubs/pull/56/commits/f8c5979fc071341ce2184d2b4c2d1a4599ba17da. One (and only one!) of the CI builds is failing though...